claim_batch/services.py

```diff
@@ -431,6 +432,207 @@ def _get_capitation_region_and_district(location_id):
+            work_data.contributions  = Premium.objects\
+                .filter(validity_from__lte = start_date)\
+                .filter(validity_from__gte = end_date)\
+                .filter(pocess_stamp_lte = end_date  )\
+                .filter(pocess_stamp_gte = end_date  )\                
+                .filter(validity_to__isnull = True)\
+                .filter(policy__product = product)\
+                .select_related('policy')
+
+            # 2 batchRun pre-calculation 
+
+            # update the service and item valuated amount 
+            claim_batch_valuation(work_data.items, work_data.services , work_data.allocated_contributions)
+
+            # 2.1 [for the futur if there is any need ]filter a calculation valid for batchRun with context BatchPrepare (got via 0.2): like allocated contribution
+            #if work_data.paymentplans:
+            #    for paymentplan in work_data.paymentplans:
+            #        if paymentplan.calculation.active_for_object(batch_run, 'BatchPrepare'):
+            #            # work_data gets updated with relevant data such as work_data.allocated_contribution
+            #            paymentplan.calculation.calculate(work_data, 'BatchPrepare'))
+            #        # 3 Generate Batchvaluation:
+            #        # 3.1 filter a calculation valid for batchRun with context BatchValuation (got via 0.2)
+            #        if paymentplan.calculation.active_for_object(batch_run, 'BatchValuate'):
+            #        # 3.2 Execute the calculation per Item or service (not claims)
+            #            paymentplan.calculation.calculate(work_data, 'BatchValuate'))
+            #else:
+            #    logger.info("no payment plan product found for product  %s for %s/%s", product_id, period, year)
+            # 4 update the claim Total amounts if all Item and services got "valuated"
+            claims = Claim.objects\
+                .filter(validity_from__lte = start_date)\
+                .filter(validity_from__gte = end_date)\
+                .filter(validity_to__isnull = True)\
+                .filter(claim_pocess_stamp_lte = end_date  )\
+                .filter(claim_pocess_stamp_gte = end_date  )\
+                .filter(Q(items__product_in = products) & Q(items__legacy_id__isnull = True) | Q(services__product_in = products) & Q(services__legacy_id__isnull = True))\
+                .prefetch_related(Prefetch('items', queryset=ClaimItem.objects.filter(legacy_id__isnull=True)))\
+                .prefetch_related(Prefetch('services', queryset=ClaimService.objects.filter(legacy_id__isnull=True)))
+            for claim in claims:
+                remunerated_amount = 0
+                for service in claim.services:
+                    remunerated_amount += service.remunerated_amount
+                for item in claim.items:
+                    remunerated_amount += item.remunerated_amount
```

Comment:
@delcroip Where this 'remunerated_amount' is used later? Can you give me an example?

Reply:
I see this comment '# 4 update the claim Total amounts if all Item and services got "valuated"'
I assume we need to add such line to update 'valuated' in Claim object, don't we (see below)?
` claim.valuated = remunerated_amount `

Reply:
no, it is a fields of the claim (<> than the claim.valuated), I just forget the save()